src/common/arrow/arrow_converter.cpp

```diff
@@ -88,14 +73,38 @@ void SetArrowMapFormat(DuckDBArrowSchemaHolder &root_holder, ArrowSchema &child,
 	InitializeChild(root_holder.nested_children.back()[0], root_holder);
 	child.children = &root_holder.nested_children_ptr.back()[0];
 	child.children[0]->name = "entries";
-	SetArrowFormat(root_holder, **child.children, ListType::GetChildType(type), options);
+	SetArrowFormat(root_holder, **child.children, ListType::GetChildType(type), options, context);
+}
+
+bool SetArrowExtension(DuckDBArrowSchemaHolder &root_holder, ArrowSchema &child, const LogicalType &type,
+                       ClientContext &context) {
```

Comment:
Me having fun here, could this be moved from `ClientContext &` to 
`optional_ptr<ClientContext>`?

And with ClientContext being passed arrow extensions are not considered?